Add Kyverno CLI manifest

dhananjaysharma2004 opened the pull request:
I have created a manifest file for Kyverno CLI in the Scoop "Main" bucket which includes version info and download link for the Kyverno CLI and follows Scoop's manifest structure.

<!-- or -->
Relates to _https://github.com/kyverno/kyverno/issues/4322_

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

aliesbelik commented:
Hi @dhananjaysharma2004.
Thank you for PR.

Could you please all the required points in [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md) first ([For Scoop buckets](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md#for-scoop-buckets)) and fix proposed manifest accordingly?

As well you can look for existent `kyverno-cli` manifests in other buckets, e.g.
https://github.com/hoilc/scoop-lemon/blob/master/bucket/kyverno-cli.json
https://github.com/okibcn/ScoopMaster/blob/master/bucket/kyverno-cli.json

dhananjaysharma2004 commented:
Hey @aliesbelik .                                                                                                                                                                                              I have checked all the required points in [contributing guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md) and proposed manifests and according to the existent `kyverno-cli` manifests in other buckets, I have made changes. Kindly check it, looking for your feedback, thanks!

aliesbelik commented on bucket/kyverno-cli.json:
```diff
@@ -0,0 +1,27 @@
+{     
```
```suggestion
{
```

Unnecessary whitespaces.

aliesbelik commented on bucket/kyverno-cli.json:
```diff
@@ -0,0 +1,27 @@
+            "url": "https://github.com/kyverno/kyverno/releases/download/v1.10.3/kyverno-cli_v1.10.3_windows_x86_64.zip",     
+            "hash": "bcc11df1a17377357705d28e84be0f660fc9c43cae30a4440ade0aec5ee298be"     
+        }
+    },     
```
```suggestion
    },
```

Unnecessary whitespaces.

aliesbelik commented on bucket/kyverno-cli.json:
```diff
@@ -0,0 +1,27 @@
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/kyverno/kyverno/releases/download/v1.10.3/kyverno-cli_v1.10.3_windows_x86_64.zip",     
+            "hash": "bcc11df1a17377357705d28e84be0f660fc9c43cae30a4440ade0aec5ee298be"     
```
```suggestion
            "hash": "bcc11df1a17377357705d28e84be0f660fc9c43cae30a4440ade0aec5ee298be"
```

Unnecessary whitespaces.

aliesbelik commented on bucket/kyverno-cli.json:
```diff
@@ -0,0 +1,27 @@
+            "hash": "bcc11df1a17377357705d28e84be0f660fc9c43cae30a4440ade0aec5ee298be"     
+        }
+    },     
+    "bin": "kyverno.exe",          
```
```suggestion
    "bin": "kyverno.exe",
```

Unnecessary whitespaces.

aliesbelik commented on bucket/kyverno-cli.json:
```diff
@@ -0,0 +1,27 @@
+{     
+    "version": "v1.10.3",
+    "description": "The Kyverno Command Line Interface (CLI) is designed to validate and test policy behavior to resources prior to adding them to a cluster.", 
```
```suggestion
    "description": "Kyverno CLI to manage Kubernetes policies", 
```

aliesbelik commented on bucket/kyverno-cli.json:
```diff
@@ -0,0 +1,27 @@
+    "description": "The Kyverno Command Line Interface (CLI) is designed to validate and test policy behavior to resources prior to adding them to a cluster.", 
+    "homepage": "https://kyverno.io/docs/kyverno-cli/",
+    "license": "Apache-2.0",
+    "notes": "Kyverno CLI for managing Kubernetes policies.",
```
```suggestion
```

Duplicates description.

aliesbelik reviewed (changes requested):
Please fix `version` and remove trailing whitespaces across the manifest.
You can use automatic checks executed for manifest file to detect possible issues.

dhananjaysharma2004 commented:
Hey @aliesbelik , thanks for your guidance. Could you please guide me in solving the following issues also 

> Error: [-] files do not contain leading UTF-8 BOM

> Error: [-] files have no lines containing trailing whitespace     

 I added trailing whitespaces in my commit [d18d49a](https://github.com/ScoopInstaller/Main/pull/5115/commits/d18d49a2ead63dd107006e32c234c951a2b3920d) but it gave another error so I removed it in  [a14a0ed](https://github.com/ScoopInstaller/Main/pull/5115/commits/a14a0edfd0e4dad9811aa20aa39bad9daeb3bddf). Kindly guide me through this, looking for your feedback, thanks!

rashil2000 commented:
Please do not modify the settings file

dhananjaysharma2004 commented:
> Please do not modify the settings file

Hey @rashil2000 , I have maintained the settings file like it was before but could you tell me another alternative to insert UTF-8 BOM character at the begining of `kyverno-cli.json `

aliesbelik commented on bucket/kyverno-cli.json:
```diff
@@ -0,0 +1,26 @@
+            "hash": "bcc11df1a17377357705d28e84be0f660fc9c43cae30a4440ade0aec5ee298be"
+        }
+    },
+    "bin": "kyverno.exe",          
```
```suggestion
    "bin": "kyverno.exe",
```
I guess you could double check first before requesting review with the same error.

aliesbelik commented:
> Hey @rashil2000 , I have maintained the settings file like it was before but could you tell me another alternative to insert UTF-8 BOM character at the begining of `kyverno-cli.json `

Why do you want to submit UTF with BOM, if the linter's error message clearly points that is an issue?

> Error: [-] files do not contain leading UTF-8 BOM 1.59s (1.58s|9ms)
> Message
>   RuntimeException: The following files have utf-8 BOM: 
>   D:\a\Main\Main\my_bucket\bucket\kyverno-cli.json

Check your encoding settings, ensure that your editor doesn't insert leading BOM or remove it before commit in preferred manner.

dhananjaysharma2004 commented:
@aliesbelik All checks have passed!

aliesbelik commented:
/verify

eddycharly commented:
Thanks for that @dhananjaysharma2004 and @aliesbelik 

Quick question, do we need to open a new PR for every release or is there some kind of automation we can use for that ?

dhananjaysharma2004 commented:
The process of releasing new versions involves updating this manifest file with information about the new release, such as the version number, download links, and hash values. So, I think we need to open a new PR for every release. @aliesbelik kindly tell if there's some kind of automation for that.

aliesbelik commented:
> So, I think we need to open a new PR for every release.

No, (normally) you don't: [App Manifest Autoupdate](https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate#testing-and-running-autoupdate).
Manifests in the bucket are updated reqularly via github action runs, using info from `autoupdate` section in the manifests themselves (if they have this section).
Another PR with updates will be required if, e.g., archives/binaries names or version/release notation will be changed, and autoupdate template becomes inactual.

dhananjaysharma2004 commented:
Ok I have used `autoupdate` section in this manifest so it is already automated.

eddycharly commented:
Awesome !

Thanks again @dhananjaysharma2004 and @aliesbelik

dhananjaysharma2004 commented:
> Awesome !
> 
> Thanks again @dhananjaysharma2004 and @aliesbelik

Your welcome @eddycharly ! I think now we can close [#4322](https://github.com/kyverno/kyverno/issues/4322)

dhananjaysharma2004 commented:
Thanks @aliesbelik ! Kindly merge this PR so we can close [#4322](https://github.com/kyverno/kyverno/issues/4322)

eddycharly commented:
@aliesbelik what are the next steps to get this in ?

aliesbelik commented:
@eddycharly I'm not a maintainer of this repo but only occasional contributor.
As for me your manifest looks good at the moment.
The only thing you could enhance is changing title of PR [as per Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md#for-scoop-buckets).
If there is any urgency with merging PR please contact maintainers.

eddycharly commented:
Nothing urgent, I just wanted to make sure we didn't miss anything expected, thanks !
